reactfire/firestore/index.tsx

```diff
@@ -49,6 +50,23 @@ export function useFirestoreDoc<T = unknown>(
   );
 }
 
+/**
+ * Get a firestore document and don't subscribe to changes
+ *
+ * @param ref - Reference to the document you want to get
+ * @param options
+ */
+export function useFirestoreDocOnce<T = unknown>(
+  ref: firestore.DocumentReference,
+  options?: ReactFireOptions<T>
+): T extends {} ? T : firestore.DocumentSnapshot {
+  return useObservable(
+    doc(ref).pipe(first()),
```

Comment:
todo: @davideast mentioned that there may be a problem with `first` never unsubscribing if the document doesn't exist. But not 100% sure

Reply:
I looked this up and I _think_ that first (as opposed to take(1)) throws an error if the stream completes with no values - https://swalsh.org/blog/rxjs-first-vs-take1 - but I'm a little new to observables, so I could be wrong. Might be something that I could write into a test!

Reply:
Just added another test (for useFirestoreDocOnce, since rxfire returns empty objects for docData) using a non-existent doc reference... seeeems to work as intended, but I am not sure how I could get inside the hooks and work out if the observable is actually unsubscribed?

See commit below.

Reply:
Makes sense to me! We were trying to remember whether `take(1)` or `first` was the problematic one. Thanks for following up!